### PR TITLE
Update trace api according to spec v0.5.0

### PR DESCRIPTION
**Pull request by kirugan**
Issue #1228

### CODE REVIEW

File: rpc/handlers.go [hunk 1]
```diff
@@ -1229,10 +1229,10 @@ func (h *Handler) SimulateTransactions(id BlockID, transactions []BroadcastedTra
...
 }
 
-func (h *Handler) TraceBlockTransactions(blockHash felt.Felt) ([]TracedBlockTransaction, *jsonrpc.Error) {
-	block, err := h.bcReader.BlockByHash(&blockHash)
-	if err != nil {
-		return nil, ErrInvalidBlockHash
+func (h *Handler) TraceBlockTransactions(id BlockID) ([]TracedBlockTransaction, *jsonrpc.Error) {
+	block, err := h.blockByID(&id)
+	if block == nil || err != nil {
+		return nil, ErrBlockNotFound
```
Comment (kirugan) [inline rpc/handlers.go:1235]: Even as the spec still says there should be "invalid block hash" error I think it doesn't make sense anymore. Will propose this change to the spec.

**Comment by omerfirmak** [comment]
I am waiting for https://github.com/NethermindEth/juno/pull/1261 to be merged before merging this just an FYI

File: rpc/handlers.go [hunk 2]
```diff
@@ -1226,10 +1226,10 @@ func (h *Handler) SimulateTransactions(id BlockID, transactions []BroadcastedTra
 	return result, nil
 }
 
-func (h *Handler) TraceBlockTransactions(blockHash felt.Felt) ([]TracedBlockTransaction, *jsonrpc.Error) {
-	block, err := h.bcReader.BlockByHash(&blockHash)
-	if err != nil {
-		return nil, ErrInvalidBlockHash
+func (h *Handler) TraceBlockTransactions(id BlockID) ([]TracedBlockTransaction, *jsonrpc.Error) {
+	block, err := h.blockByID(&id)
+	if block == nil || err != nil {
```
Comment (omerfirmak) [inline rpc/handlers.go:1231]: why do we need the `block == nil` now?

**Review by omerfirmak — APPROVED** [review]

**Comment by omerfirmak** [comment]
@kirugan feel free to merge after taking a look at the changes I force pushed.

**Pull request merged**